src/txn/txn_ckpt.c

```diff
@@ -627,6 +629,22 @@ __checkpoint_prepare(WT_SESSION_IMPL *session, bool *trackingp, const char *cfg[
 
     __wt_writeunlock(session, &txn_global->rwlock);
 
+    /*
+     * Open an internal session to allocate a transaction id that will be used for removing history
+     * entries. This is used when a prepare transaction rollback occurs in parallel to a checkpoint.
+     * Ensure that this transaction id is published before taking the checkpoint's snapshot.
+     */
+    WT_RET(__wt_open_internal_session(conn, "checkpoint internal", true, session->flags,
```

Comment:
Doesn't this have to be after `__wt_txn_bump_snapshot` which is where checkpoint takes a newer snapshot?

Reply:
Can we get a comment explaining why the checkpoint snap_max isn't sufficient?

Reply:
Added a comment to explain the need of using a dedicated transaction id instead of a snap_max.